### ISSUE TITLE
How to prepare my data?

### ISSUE THREAD

**Issue by qinxiaoyi** [gogyzzz/heymann-nn-gev-bf#1]
I prepare my data for training,but the result is not good.
I have 6ch and use AISHELL data for train. Firstly, I simulation the reverberation and noise audio use rir-generator and add-noise method. Then,the unreverberated audios obtained by using the dereverberation method，the unreverberated audios as a noisy set,noisy.scp。the Raw clean speech is clean.scp
my trainset is fellow,
clean.6ch.scp
SSB00010005 /home/qinxy/simulation_far_field_data/AISHELL-ASR-SSB1-200/SSB0001/SSB00010005.wav
SSB00010005 /home/qinxy/simulation_far_field_data/AISHELL-ASR-SSB1-200/SSB0001/SSB00010005.wav
SSB00010005 /home/qinxy/simulation_far_field_data/AISHELL-ASR-SSB1-200/SSB0001/SSB00010005.wav
SSB00010005 /home/qinxy/simulation_far_field_data/AISHELL-ASR-SSB1-200/SSB0001/SSB00010005.wav
SSB00010005 /home/qinxy/simulation_far_field_data/AISHELL-ASR-SSB1-200/SSB0001/SSB00010005.wav
SSB00010005 /home/qinxy/simulation_far_field_data/AISHELL-ASR-SSB1-200/SSB0001/SSB00010005.wav
SSB00010007 /home/qinxy/simulation_far_field_data/AISHELL-ASR-SSB1-200/SSB0001/SSB00010007.wav
SSB00010007 /home/qinxy/simulation_far_field_data/AISHELL-ASR-SSB1-200/SSB0001/SSB00010007.wav
SSB00010007 /home/qinxy/simulation_far_field_data/AISHELL-ASR-SSB1-200/SSB0001/SSB00010007.wav
…………

noisy.6ch.scp is fellow,
SSB00010005 /home/qinxy/simulation_far_field_data/data/SSB00010005_0.wav
SSB00010005 /home/qinxy/simulation_far_field_data/data/SSB00010005_1.wav
SSB00010005 /home/qinxy/simulation_far_field_data/data/SSB00010005_2.wav
SSB00010005 /home/qinxy/simulation_far_field_data/data/SSB00010005_3.wav
SSB00010005 /home/qinxy/simulation_far_field_data/data/SSB00010005_4.wav
SSB00010005 /home/qinxy/simulation_far_field_data/data/SSB00010005_5.wav
SSB00010007 /home/qinxy/simulation_far_field_data/data/SSB00010007_0.wav
SSB00010007 /home/qinxy/simulation_far_field_data/data/SSB00010007_1.wav
…………
The number after the file name represents the channel.

Then, i use ff.py for training.
Is there any question of my steps?

**Comment by gogyzzz** [comment]
Your method is right. 
I started this project because there is some wrong technique(concatenating all wave sequence, ... etc.) in the original code(heymann's).
Unfortunately, I've stopped this project for my schedule. 
Sorry...

**Comment by qinxiaoyi** [comment]
Thank you for your answer, i use your model for test and get a good result in far-field data. But my model is bad. I'm sorry you stopped the project.
I guess maybe my simulation data is not match with raw clean speech.
Your code have great help for me. 
Best wish!

**Comment by jacktong1989** [comment]
I managed to run the code. However, the enhanced spectra are almost the same as the original spectra, even when I use the model and test data provided by the author. Can someone give me a reason?

**Comment by gogyzzz** [comment]
@jacktong1989 
Did you train with paper's parameter? (learning rate, etc.)

**Comment by jacktong1989** [comment]
Thanks for your timely reply. I do not have access to the original complete chime3 dataset yet.  I only have the isolated part of chime3 dataset so I cannot train a model for chime 3.

Alternatively, I used the neural network model provided in the folder "sample_exp/best_state_dic.pth" and used my own far-field reverberant sentence for testing. The results almost remain the same after being processed. That is the problem.

Maybe the NN model you trained is not suitable for my task. May I ask how you obtain the NN model in "sample_exp/best_state_dic.pth" ? What kind of data did you use to obtain that model?

**Comment by jacktong1989** [comment]
I have met the same problem as qinxiaoyi mentioned above.  I use your model for test and get a good result in far-field data. But the result on my own model is bad. I trained my own model using reverberated speech in a quiet meeting room as clean data  and using air conditioners as noise data. Is there any problem in preparing training data like this? By the way, how did you prepare the training data in your project? Thanks in advance.

**Comment by gogyzzz** [comment]
@jacktong1989 
Oh, the sample_exp/best_state_dic.pth file is just for testing my algorithm.
It trained on just 2 wav samples.
And I wonder how my model got a good result on your far-field data.
How was your test data recorded? 
The environment mismatch can be one of the reasons.

**Comment by jacktong1989** [comment]
My data was recorded in a medium-sized meeting room. I found the model you trained on 2 sentences is indeed slightly better than my model trained on 8GB real data acquired in a meeting room.  This really confuses me.
